### PR TITLE
Allows editing the colors of the ramp in a numeric category list

### LINKED ISSUE #7725: [Fill component] Allow to edit color ramps
pasted_image_at_2016_06_08_17_31

Note: we don't need to implement the 'save' button, since any change to the color list will be automatically sent

### PR DESCRIPTION
**Pull request by javierarce**
Closes #7725 

pasted_image_at_2016_06_08_17_31

### CODE REVIEW

**Comment by xavijam** [comment]
👀 Looks beautiful man :)

**Comment by javierarce** [comment]
@xavijam: thanks! Merged with master. Can you CR this, please?

**Comment by xavijam** [comment]
:joy: sure! _on it_

File: app/assets/stylesheets/editor-3/ramp-list-view.css.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
+@import 'cdb-utilities/mixins';
```
Comment (xavijam) [inline app/assets/stylesheets/editor-3/ramp-list-view.css.scss]: Q: do we have more css files for the fill component?

File: app/assets/stylesheets/editor-3/ramp-list-view.css.scss [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+@import 'cdb-variables/sizes';
+@import 'cdb-variables/colors';
+
+.CustomRamp {
+  display: flex;
```
Comment (xavijam) [inline app/assets/stylesheets/editor-3/ramp-list-view.css.scss]:
```css
@include display-flex();
```

File: app/assets/stylesheets/editor-3/ramp-list-view.css.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+}
+.CustomRamp .CustomRamp-list {
+  flex: 1;
```
Comment (xavijam) [inline app/assets/stylesheets/editor-3/ramp-list-view.css.scss]:
```scss
@include flex(1);
```

File: app/assets/stylesheets/editor-3/ramp-list-view.css.scss [hunk 4]
```diff
@@ -0,0 +1,17 @@
+@import 'cdb-utilities/mixins';
+@import 'cdb-variables/sizes';
```
Comment (xavijam) [inline app/assets/stylesheets/editor-3/ramp-list-view.css.scss]: We are requiring `sizes` but we are not using it, in any case i feel you can use it for the padding, with the base size, no @piensaenpixel?

File: lib/assets/javascripts/cartodb3/components/custom-list/custom-view.js [hunk 5]
```diff
@@ -86,16 +86,20 @@ module.exports = CoreView.extend({
...
       collection: this.collection,
       typeLabel: this.options.typeLabel,
       ItemView: this.options.itemView,
       itemTemplate: this.options.itemTemplate
     });
-    this.$el.append(listView.render().el);
-    listView.highlight();
-    this.addView(listView);
+    this.$el.append(this._listView.render().el);
+    this._listView.highlight();
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/custom-list/custom-view.js]: Where it `highlight` function was before?

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-ramps/input-ramp-list-view.js [hunk 6]
```diff
@@ -44,12 +70,33 @@ module.exports = CoreView.extend({
...
+  _initModels: function () {
+    this._customRamp = new cdb.core.Model();
+    this.add_related_model(this._customRamp);
+
+    this._setupCollection();
   },
 
   _initBinds: function () {
+    this.model.bind('change:bins', this._onChangeBins, this);
     this.collection.bind('change:selected', this._onSelectItem, this);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-ramps/input-ramp-list-view.js]: `this.add_related_model(this.collection)`

**Comment by xavijam** [comment]
Minor things, 👍

File: app/assets/stylesheets/editor-3/ramp-list-view.css.scss (hunk 1 above)
Comment (xavijam) [inline app/assets/stylesheets/editor-3/ramp-list-view.css.scss]: If so, it is a good moment to create a folder only for the fill component styles :).

File: app/assets/stylesheets/editor-3/ramp-list-view.css.scss (hunk 4 above)
Comment (piensaenpixel) [inline app/assets/stylesheets/editor-3/ramp-list-view.css.scss]: sure

File: app/assets/stylesheets/editor-3/ramp-list-view.css.scss (hunk 1 above)
Comment (javierarce) [inline app/assets/stylesheets/editor-3/ramp-list-view.css.scss]: Done.

File: app/assets/stylesheets/editor-3/ramp-list-view.css.scss (hunk 4 above)
Comment (javierarce) [inline app/assets/stylesheets/editor-3/ramp-list-view.css.scss]: 👌

**Pull request merged**